### ISSUE TITLE
'Error: Cannot enqueue Query after being destroyed' when calling liveDB.end

### ISSUE THREAD

**Issue by sushitommy** [numtel/meteor-mysql#24]
I got this code to test a connection:

				
			var liveDB = new LiveMysql(
			{
				host: host,
				post: port,
				user: user,
				password: password
			}, function(err, result) {
		  		liveDB.end();
			});

And I keep getting this error: 'Error: Cannot enqueue Query after being destroyed' at Protocol._validateEnqueue (C:\Users\Tommy J\AppData\Local\.meteor\packages\nu
mtel_mysql\0.1.9\npm\node_modules\mysql-live-select\node_modules\zongji\node_modules\mysql\lib\protocol\Protocol.js:199:
16)

How can I end a connection properly?

**Comment by numtel** [comment]
That callback occurs immediately after the first database connection is ready but does not wait for the Zongji binlog connection to be ready as there is no callback for that event. To get around this, wait at least a second before closing the connection:

```javascript
  var liveDb = new LiveMysql({
    ...
  }, Meteor.bindEnvironment(function(error) {
    Meteor.setTimeout(function() {
      liveDb.end();
    }, 1000);
  }));
```

Definitely a bug, I'll see if I can come up with a solution soon.

**Comment by sushitommy** [comment]
Thanks, and great work!!

**Comment by numtel** [comment]
Ok, if you upgrade to the just released 0.1.11, the callback now is not called until the zongji instance is ready and able to be shutdown.
